### PR TITLE
[SYCL] Make spelling of Debug value for CMAKE_BUILD_TYPE variable case insensitive

### PR DESCRIPTION
**Pull request by s-kanaev**

### CODE REVIEW

File: sycl/cmake/modules/AddSYCLUnitTest.cmake [hunk 1]
```diff
@@ -6,7 +6,8 @@ macro(add_sycl_unittest test_dirname link_variant)
   # Enable exception handling for these unit tests
   set(LLVM_REQUIRES_EH 1)
 
-  if (MSVC AND CMAKE_BUILD_TYPE MATCHES "Debug")
+  string(TOLOWER "${CMAKE_BUILD_TYPE}" build_type_lower)
```
Comment (bader) [inline sycl/cmake/modules/AddSYCLUnitTest.cmake:9]: I would report an error if CMAKE_BUILD_TYPE is not "Debug".

Comment (s-kanaev) [reply sycl/cmake/modules/AddSYCLUnitTest.cmake:9]: I can't agree with it.
See https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html
```
Note that configuration names are case-insensitive. The value of this variable will be the same as it is specified when invoking CMake. For instance, if -DCMAKE_BUILD_TYPE=ReLeAsE is specified, then the value of CMAKE_BUILD_TYPE will be ReLeAsE.
```

Comment (bader) [reply sycl/cmake/modules/AddSYCLUnitTest.cmake:9]: I don't your point. The documentation says that value of CMAKE_BUILD_TYPE is case-insensitive, but my point is "Let DPC++ CMake files support only one spelling version - Debug and reject all the others e.g. DeBuG".

See the same document: 
> Possible values are empty, Debug, Release, RelWithDebInfo, MinSizeRel, ...

Comment (s-kanaev) [reply sycl/cmake/modules/AddSYCLUnitTest.cmake:9]: Still, cmake isn't sensitive to case of value of `CMAKE_BUILD_TYPE` i.e. if it's `DeBuG` or `debug` etc., then cmake acts the same as if it was `Debug`.

Comment (bader) [reply sycl/cmake/modules/AddSYCLUnitTest.cmake:9]: If you prefer to support all spelling, could you set `build_type_lower` string once and re-use in both macro functions?

Comment (s-kanaev) [reply sycl/cmake/modules/AddSYCLUnitTest.cmake]: Updated.

File: sycl/cmake/modules/AddSYCLUnitTest.cmake [hunk 2]
```diff
@@ -6,7 +6,7 @@ macro(add_sycl_unittest test_dirname link_variant)
   # Enable exception handling for these unit tests
   set(LLVM_REQUIRES_EH 1)
 
-  if (MSVC AND CMAKE_BUILD_TYPE MATCHES "Debug")
+  if (MSVC AND build_type_lower MATCHES "debug")
```
Comment (bader) [inline sycl/cmake/modules/AddSYCLUnitTest.cmake:9]: Actually I thought about moving duplicated code within `sycl/cmake/modules/AddSYCLUnitTest.cmake` only.
How about adding a new macro for configurations common for both `add_sycl_unittest` and `add_sycl_unittest_with_device`?
E.g. 
```cmake 
macro(add_sycl_unittest_common CONFIG_SUFFIX)
  # Enable exception handling for these unit tests
  set(LLVM_REQUIRES_EH 1)
  string(TOLOWER "${CMAKE_BUILD_TYPE}" build_type_lower)
  if (MSVC AND build_type_lower MATCHES "debug")
    set (CONFIG_SUFFIX "d")
  endif
endmacro()

macro(add_sycl_unittest_with_device test_dirname link_variant)
  add_sycl_unittest_common(CONFIG_SUFFIX)
  set(sycl_obj_target "sycl${CONFIG_SUFFIX}_object")
  set(sycl_so_target "sycl${CONFIG_SUFFIX}")
  set(XPTI_LIB xpti${CONFIG_SUFFIX})
...
```

Comment (s-kanaev) [reply sycl/cmake/modules/AddSYCLUnitTest.cmake:9]: `add_sycl_unittest_with_device` was dismissed, so I've reverted some changes.

**Review by romanovvlad — APPROVED** [review]

**Pull request merged**